aries-site/src/examples/templates/feedback/ModalTaskFlowExample/ModalTaskFlowExample.js

```diff
@@ -4,48 +4,52 @@ import { ModalDialog } from '../../ModalDialog';
 import { config, PowerOnFlow } from './PowerOnFlow';
 import { TaskFlowFeedback } from './TaskFlowFeedback';
 
+const showFeedback = () => {
+  let result = false;
+  // A check for whether the should be shown the feedback survey
+  // could be implemented here. Solicitation of users' feedback
+  // should respect users' time and be mindful to not over sample.
+
+  // always returning true for demo purposes
+  result = true;
+  return result;
+};
+
 export const ModalTaskFlowExample = ({ containerRef }) => {
   // containerRef is for demonstration purposes on this site. Most
   // implementations should likely remove.
   const [showModal, setShowModal] = useState(true);
-  const [showFeedback, setShowFeedback] = useState(false);
   // announce when the layer opens
   const announce = useContext(AnnounceContext);
-  const defaultTitle = config.title;
-  const defaultContent = (
+  const flowTitle = config.title;
+  const flowContent = (
     <PowerOnFlow
       onCancel={() => {
         announce(`Power on device modal cancelled and closed.`, 'assertive');
         setShowModal(false);
       }}
       onClose={() => {
-        setShowFeedback(true);
+        if (showFeedback()) {
```

Comment:
I see the difference now, I'm fine with either approach -- the useEffect or this. The useEffect might be a little cleaner now that I'm seeing them both, but fine with either.

Reply:
I will revert.

Reply:
Actually, I'm going to merge. We can do that clean up later.

Reply:
👆 I was going to just tell you if we should merge then clean up in original PR